lib/swoosh/adapters/smtp.ex

```diff
@@ -40,13 +41,51 @@ if Code.ensure_loaded?(:gen_smtp_client) do
       sender = Helpers.sender(email)
       recipients = all_recipients(email)
       body = Helpers.body(email, config)
-      case :gen_smtp_client.send_blocking({sender, recipients, body}, config) do
+
+      case :gen_smtp_client.send_blocking(
+        {sender, recipients, body},
+        gen_smtp_config(config)
+      ) do
         receipt when is_binary(receipt) -> {:ok, receipt}
         {:error, type, message} -> {:error, {type, message}}
         {:error, reason} -> {:error, reason}
       end
     end
 
+    @config_transformations %{
+      port: &String.to_integer/1,
+      retries: &String.to_integer/1,
+      ssl: {&String.to_existing_atom/1, [true, false]},
```

Comment:
why are you using `to_existing_atom/1` here? (just curious)

Reply:
So the user cannot misspell the config and create a new one.
While it doesn't really matter, as it's caught immediately, I always use `to_existing_atom` if `to_atom` is not absolutely needed.

Reply:
👍

Reply:
I changed this to `to_atom` in the end:

- this is not from user input, should be safe to use `to_atom`
- using `to_existing_atom` raises earlier than my valid option check, troublesome